### PR TITLE
Table component enhancement: Details

### PR DESCRIPTION
**Pull request by wanxe**
> Sometimes a table is not enough to display all the fields. I think it is a good idea to incorporate this functionality that, in addition, emits events when closing or opening. Which opens many possibilities, such as making an ajax request to display the details of a user. Personally this utility I use daily in my work, so I've considered making this contribution to further improve this incredible library.

Table component enhancements:

- Tables now allow the option to show details using an scoped slot "detail".
- Table trigger an event when close or open each row to be able to control user actions.
- Adds this feature to the documentation.

[Video Link](https://mega.nz/#!Qh5nEDTI!koDw8kdpMWLbGvNoNfQXoKDaC8GehLWCmCgtoDgxNAo)

### CODE REVIEW

File: docs/pages/documentation/data/Table.vue [hunk 1]
```diff
@@ -79,15 +82,45 @@
                                 {{ props.row.gender }}
                             </b-table-column>
                         </template>
-
-                        <div slot="empty" class="has-text-centered">
```
Comment (rafaelpimpa) [inline docs/pages/documentation/data/Table.vue]: You removed the empty slot.

File: docs/pages/documentation/data/Table.vue [hunk 2]
```diff
@@ -42,15 +44,18 @@
...
                         :loading="isLoading"
                         :mobile-cards="hasMobileCards"
                         :paginated="isPaginated"
                         :per-page="perPage"
                         :pagination-simple="isPaginationSimple"
+                        default-sort="user.first_name"
                         :selected.sync="selected"
-                        :checked-rows.sync="checkedRows"
-                        default-sort="user.first_name">
+                        @details-opened="onOpenedDetail"
```
Comment (rafaelpimpa) [inline docs/pages/documentation/data/Table.vue]: Better rename to 'details-open' and 'details-close' to keep like all others events.

File: docs/pages/documentation/data/Table.vue [hunk 3]
```diff
@@ -10,10 +10,12 @@
             <b-switch v-model="isNarrowed">Narrowed</b-switch>
             <b-switch v-model="isCheckable">Checkable</b-switch>
             <b-switch v-model="isLoading">Loading state</b-switch>
+        </div>
+        <div class="block">
             <b-switch v-model="isPaginated">Paginated</b-switch>
             <b-switch v-model="isPaginationSimple">Simple pagination</b-switch>
-            <b-switch v-model="isEmpty">Empty</b-switch>
```
Comment (rafaelpimpa) [inline docs/pages/documentation/data/Table.vue]: You removed the empty switch.

File: .gitignore [hunk 4]
```diff
@@ -6,6 +6,5 @@ lib/
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
-test/unit/coverage
-test/e2e/reports
-selenium-debug.log
```
Comment (rafaelpimpa) [inline .gitignore]: Any reason to remove these ones?

Comment (wanxe) [inline .gitignore]: Probably was erased it by fixing the conflicts. I will add them again.

File: docs/pages/documentation/data/Table.vue (hunk 2 above)
Comment (wanxe) [inline docs/pages/documentation/data/Table.vue]: Ok fair enought.

**Comment by wanxe** [comment]
Ok, I hope fix this PR this weekend. Thank you Rafael! :)

**Comment by wanxe** [comment]
I think is already fixed 🎉

**Comment by wanxe** [comment]
I added a new prop to allow changing the icon pack.
Also fixes missed the hasDetails prop documentation and the missed.
I think that's all for the moment :)

File: docs/pages/documentation/data/Table.vue [hunk 5]
```diff
@@ -353,7 +353,22 @@
...
+                    },
+                    {
+                        name: '<code>hasDetails</code>',
+                        description: 'Allows/disallows row details. (Check scoped slots documentation)',
+                        type: 'Boolean',
+                        values: '—',
+                        default: 'false'
+                    },
+                    {
+                        name: '<code>iconPack</code>',
```
Comment (jtommy) [inline docs/pages/documentation/data/Table.vue]: IconPack prop is not required

File: src/components/table/Table.vue [hunk 6]
```diff
@@ -72,8 +73,8 @@
                         <td v-if="hasDetails">
                             <button class="is-transparent" @click="toggleDetails(row)">
                                 <b-icon :class="{'expanded': isVisibleDetailRow(row)}"
-                                        pack="fa"
-                                        icon="arrow-right"
+                                        :pack="iconPack"
```
Comment (jtommy) [inline src/components/table/Table.vue]: pack it isn't required. You can use "both" attribute (see thead sort icon)

File: src/components/table/Table.vue [hunk 7]
```diff
@@ -172,12 +173,22 @@
                 type: Boolean,
                 default: false
             },
+            iconPack: {
```
Comment (jtommy) [inline src/components/table/Table.vue]: Remove

File: src/components/table/Table.vue [hunk 8]
```diff
@@ -172,12 +173,22 @@
...
+                type: String,
+                default: 'fa'
+            },
             rowClass: {
                 type: Function,
                 default: () => ''
             }
         },
         data() {
+            let chevronIcon
```
Comment (jtommy) [inline src/components/table/Table.vue]: Remove (next time you can use a computed property to do it).

**Comment by wanxe** [comment]
Thank you @jtommy for the icon tip!  
I've squashed the commits into one with the changes.

**Comment by rafaelpimpa** [comment]
Hey @wanxe, can you fix the conflict? That file (package-lock) is not necessary to commit.

**Comment by wanxe** [comment]
Done mate!

**Comment by rafaelpimpa** [comment]
Your PR now wants to delete the file 😅

**Comment by wanxe** [comment]
Lol sorry man, check now plz.

**Comment by rafaelpimpa** [comment]
Thanks! I'm gonna make some adjustments but the rest is great!

**Comment by rafaelpimpa** [comment]
I see that you added a ``hasDetail`` prop to column, but didn't found anything on source, did you forget to remove from the docs?

**Comment by wanxe** [comment]
I do not think so, at least in my code. 
As you can see the prop is on the Table.vue source code.
screen shot 2017-07-27 at 16 30 39

**Pull request merged**